### PR TITLE
glew 2.1.0

### PR DESCRIPTION
**Pull request by ericLemanissier**
Specify library name and version:  **glew/2.1.0**

this is a port of https://github.com/bincrafters/conan-glew

- [X] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [X] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [X] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [X] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/glew/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("%s-%s" % (self.name, self.version), self._source_subfolder)
+        if "patches" in self.conan_data:
+            for patch in self.conan_data["patches"][self.version]:
+                tools.patch(**patch)
+#        tools.replace_in_file("%s/build/cmake/CMakeLists.txt" % self._source_subfolder, "include(GNUInstallDirs)",
```
Comment (SSE4) [inline recipes/glew/all/conanfile.py:33]: remove dead code

File: recipes/glew/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+#"""
+#include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+#conan_basic_setup()
+#include(GNUInstallDirs)
+#""")
+
+    def _configure_cmake(self):
+        cmake = CMake(self)
+        cmake.definitions["BUILD_UTILS"] = "OFF"
+        cmake.definitions["CONAN_GLEW_DEFINITIONS"] = ";".join(self._glew_defines)
```
Comment (SSE4) [inline recipes/glew/all/conanfile.py:43]: is this actually used anywhere?

Comment (ericLemanissier) [reply recipes/glew/all/conanfile.py]: it used to be in https://github.com/bincrafters/conan-glew/blob/581c58940d8c6d09abc74d6f4dcb1ef41cbc8902/FindGLEW.cmake.in#L21, but that won't make it in CCI !

File: recipes/glew/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+        self.copy("include/*", ".", "%s" % self._source_subfolder, keep_path=True)
+        self.copy("%s/license*" % self._source_subfolder, dst="licenses",  ignore_case=True, keep_path=False)
+        self.copy(pattern="LICENSE", dst="licenses", src=self._source_subfolder)
+
+        if self.settings.os == "Windows":
+            if self.settings.compiler == "Visual Studio":
+                if self.options.shared:
+                    if self.settings.build_type == "Release":
+                        self.copy("glew32.lib", "lib", "lib", keep_path=False)
```
Comment (SSE4) [inline recipes/glew/all/conanfile.py:61]: is it copying into the same folder? I feel like most of these lines are no longer needed as we use install, maybe we should just set up `cpp_info.bindis` or `cpp_info.libdirs`, but I think we probably should just package everything as install does, avoid redundant repackaging here

File: recipes/glew/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+            if self.settings.compiler == "Visual Studio":
+                if not self.options.shared:
+                    self.cpp_info.libs[0] = "lib" + self.cpp_info.libs[0]
+
+        else:
+            self.cpp_info.libs = ['GLEW']
+
+        if self.settings.build_type == "Debug":
+            self.cpp_info.libs[0] += "d"
```
Comment (p-podsiadly) [inline recipes/glew/all/conanfile.py:69]: This can be replaced by `tools.collect_libs`

File: recipes/glew/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    name = "glew"
+    description = "The GLEW library"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "http://github.com/nigels-com/glew"
+    topics = ("conan", "glew", "opengl", "wrangler", "loader", "binding")
+    license = "MIT"
+    exports_sources = ["CMakeLists.txt", "patches/*"]
+    generators = "cmake"
+    settings = "os", "arch", "build_type", "compiler"
+    options = {"shared": [True, False]}
```
Comment (uilianries) [inline recipes/glew/all/conanfile.py:15]: Why fPIC is not listed?

File: recipes/glew/all/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+    default_options = {"shared": False}
+    _source_subfolder = "source_subfolder"
```
Comment (SpaceIm) [inline recipes/glew/all/conanfile.py:17]:
```suggestion
    
    _cmake = None

    @property
    def _source_subfolder(self):
        return "source_subfolder"

    @property
    def _build_subfolder(self):
        return "build_subfolder"
```

File: recipes/glew/all/conanfile.py [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("%s-%s" % (self.name, self.version), self._source_subfolder)
+        if "patches" in self.conan_data:
+            for patch in self.conan_data["patches"][self.version]:
+                tools.patch(**patch)
```
Comment (SpaceIm) [inline recipes/glew/all/conanfile.py:32]:
```suggestion
```

File: recipes/glew/all/conanfile.py [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+
+    def _configure_cmake(self):
+        cmake = CMake(self)
+        cmake.definitions["BUILD_UTILS"] = "OFF"
+        cmake.configure()
+        return cmake
```
Comment (SpaceIm) [inline recipes/glew/all/conanfile.py:38]:
```suggestion
        if self._cmake:
            return self._cmake
        self._cmake = CMake(self)
        self._cmake.definitions["BUILD_UTILS"] = "OFF"
        self._cmake.configure(build_folder=self._build_subfolder)
        return self._cmake
```

File: recipes/glew/all/conanfile.py [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+
+    def build(self):
+        cmake = self._configure_cmake()
```
Comment (SpaceIm) [inline recipes/glew/all/conanfile.py:41]:
```suggestion
        for patch in self.conan_data.get("patches", {}).get(self.version, []):
            tools.patch(**patch)
        cmake = self._configure_cmake()
```

File: recipes/glew/all/test_package/CMakeLists.txt [hunk 10]
```diff
@@ -0,0 +1,8 @@
+cmake_minimum_required(VERSION 3.0)
```
Comment (SpaceIm) [inline recipes/glew/all/test_package/CMakeLists.txt:1]:
```suggestion
cmake_minimum_required(VERSION 2.8.12)
```

File: recipes/glew/all/conanfile.py (hunk 5 above)
Comment (ericLemanissier) [reply recipes/glew/all/conanfile.py:15]: IDK, but you're not the only who noticed it https://github.com/bincrafters/community/issues/1158 !

File: recipes/glew/all/conanfile.py [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+        cmake.install()
+
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+        for pdb_file in glob.glob(os.path.join(self.package_folder, "lib", "*.pdb")):
+            os.remove(pdb_file)
+
+        self.copy(pattern="LICENSE.txt", dst="licenses", src=self._source_subfolder)
+
+    def package_info(self):
```
Comment (SpaceIm) [inline recipes/glew/all/conanfile.py:77]:
```suggestion
    def package_info(self):
        self.cpp_info.names["cmake_find_package"] = "GLEW"
        self.cpp_info.names["cmake_find_package_multi"] = "GLEW"
```
cmake_find_package: https://cmake.org/cmake/help/v3.14/module/FindGLEW.html
cmake_find_package_multi: namespace is also GLEW for cmake_find_package_multi, but I think that target is GLEW::glew or GLEW::glew_s :s

Comment (ericLemanissier) [reply recipes/glew/all/conanfile.py:77]: shouldn't we use the component feature to set the target name `GLEW::glew` or `GLEW::glew_s` based on shared option ? Or maybe what you are proposing is already creating the `GLEW::GLEW` target, which would be conform to https://cmake.org/cmake/help/latest/module/FindGLEW.html (sorry I don't really know the components feature of conan yet, and what they allow to do)

Comment (prince-chrismc) [inline recipes/glew/all/conanfile.py:77]: Yes it guys are both saying the same time, should look something like this, but I am not sure what the lib names are

```suggestion
    def package_info(self):
        self.cpp_info.names["cmake_find_package"] = "GLEW"
        self.cpp_info.names["cmake_find_package_multi"] = "GLEW"
        if self.options.shared:
                self.cpp_info.components["glew"].libs = ["glew"] // Correct lib name
         else:
                self.cpp_info.components["glew_s"].libs = ["glew"] // Correct lib name
```

File: recipes/glew/all/conanfile.py [hunk 12]
```diff
@@ -0,0 +1,80 @@
...
+        if self.settings.os == "Windows" and not self.options.shared:
+            self.cpp_info.defines.append("GLEW_STATIC")
```
Comment (prince-chrismc) [inline recipes/glew/all/conanfile.py:79]:
```suggestion
           self.cpp_info.components["glew_s"].defines.append("GLEW_STATIC")
```

File: recipes/glew/all/conanfile.py (hunk 11 above)
Comment (SpaceIm) [reply recipes/glew/all/conanfile.py:77]: Something like this:
```python
    def package_info(self):
        self.cpp_info.names["cmake_find_package"] = "GLEW"
        self.cpp_info.names["cmake_find_package_multi"] = "GLEW"
        self.cpp_info.components["glewlib"].names["cmake_find_package"] = "GLEW"
        self.cpp_info.components["glewlib"].names["cmake_find_package_multi"] = "glew" if self.options.shared else "glew_s"
        if self.settings.os == "Windows" and not self.options.shared:
            self.cpp_info.components["glewlib"].defines.append("GLEW_STATIC")
```

**Comment by daniellecandid** [comment]
Sorry if this is against form, but hoping to ping on this. Would love to use this package! @danimtb

**Pull request merged**